### PR TITLE
OADP-823: Address oadp-1.0.z with ocp 4.11 csi compatibility issue

### PR DESCRIPTION
**Pull request by shubham-pampattiwar**

### CODE REVIEW

File: controllers/validator.go [hunk 1]
```diff
@@ -100,6 +103,30 @@ func (r *DPAReconciler) ValidateVeleroPlugins(log logr.Logger) (bool, error) {
...
+		// check csi compatibility with cluster version
+		// velero version <1.9 expects API group snapshot.storage.k8s.io/v1beta1, while OCP 4.11 (k8s 1.24) has only snapshot.storage.k8s.io/v1
+		for _, plugin := range dpa.Spec.Configuration.Velero.DefaultPlugins {
+			if plugin == oadpv1alpha1.DefaultPluginCSI {
+				clusterVersion, err := getClusterVersion()
+				if err != nil {
+					return false, err
+				}
+
+				if clusterVersion.Major == "1" && clusterVersion.Minor >= "24" {
```
Comment (weshayutin) [inline controllers/validator.go:116]: looks good! thank you

**Comment by weshayutin** [comment]
looks good to me..  I suspect we should add a unit test for this issue.  Should be easy enough and can be in a follow up. 
Thanks

File: controllers/validator.go (hunk 1 above)
Comment (kaovilai) [reply controllers/validator.go:116]: Why are we doing nested for loops for no reason? The outerloop is already for each plugin. I would put this logic in line 90.
https://github.com/openshift/oadp-operator/blob/65159dd313799cc3ca52a0a4b31623098ee62900/controllers/validator.go#L89-L120

Comment (sseago) [reply controllers/validator.go:116]: This works until k8s 2.0, but I can only assume that oadp 1.0 won't be supported at all on whatever version of OpenShift that would be.

**Review by kaovilai — CHANGES_REQUESTED** [review]
Please don't do 
```
for _, plugin := range dpa.Spec.Configuration.Velero.DefaultPlugins {
  for _, plugin := range dpa.Spec.Configuration.Velero.DefaultPlugins {
  }
}
```

**Review by sseago — APPROVED** [review]
lgtm

File: controllers/validator.go (hunk 1 above)
Comment (shubham-pampattiwar) [reply controllers/validator.go:116]: @kaovilai @sseago earlier I had a function for this and then relocated the logic, copy/paste oversight, will rectify, Thank you !

File: controllers/validator.go [hunk 2]
```diff
@@ -100,6 +103,30 @@ func (r *DPAReconciler) ValidateVeleroPlugins(log logr.Logger) (bool, error) {
...
+					return false, errors.New("when using csi, velero version <1.9 expects API group snapshot.storage.k8s.io/v1beta1, while OCP 4.11+ (k8s 1.24+) has only snapshot.storage.k8s.io/v1, please disable CSI support, or if CSI is needed, upgrade to OADP 1.1+")
```
[the remark this answers is not in the document]
Comment (kaovilai) [reply controllers/validator.go]:
```suggestion
					return false, errors.New("csi plugin on OADP 1.0 (velero version <1.9) requires API snapshot.storage.k8s.io/v1beta1. On OCP 4.11+ (k8s 1.24+), If CSI is needed, upgrade to OADP 1.1+")
```

[the remark this answers is not in the document]
Comment (shubham-pampattiwar) [reply controllers/validator.go]: @eemcmullan the suggested error message makes it seem like csi is not supported with prior velero 1.9 and priror OCP versions, which is not the case right ? We need to tell the actual reason imo. but yeah agreed on making concise, any more suggestions with the reason embedded in error message ?

File: controllers/validator.go (hunk 1 above)
Comment (shubham-pampattiwar) [reply controllers/validator.go]: Done, updated the PR !

**Review by kaovilai — APPROVED** [review]
Agree with Emily on condensing error string a bit. otherwise
/lgtm

**Review by sseago — APPROVED** [review]

**Review by sseago — APPROVED** [review]

**Review by kaovilai — APPROVED** [review]

**Pull request merged**